prometheus-collector/src/main/java/io/opentelemetry/contrib/prometheus/PrometheusCollector.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.prometheus;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.metrics.InstrumentType;
+import io.opentelemetry.sdk.metrics.data.AggregationTemporality;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.CollectionRegistration;
+import io.opentelemetry.sdk.metrics.export.MetricReader;
+import io.opentelemetry.sdk.metrics.internal.export.MetricProducer;
+import io.prometheus.client.Collector;
+import io.prometheus.client.CollectorRegistry;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * A reader of OpenTelemetry metrics that exports into Prometheus as a Collector.
+ *
+ * <p>Usage: <code>sdkMeterProvider.registerMetricReader(PrometheusCollector.create());</code>
+ */
+public final class PrometheusCollector implements MetricReader {
+
+  private final Collector collector;
+  private volatile MetricProducer metricProducer = MetricProducer.noop();
+
+  PrometheusCollector() {
+    this.collector = new CollectorImpl(() -> getMetricProducer().collectAllMetrics());
+    this.collector.register();
+  }
+
+  /**
+   * Returns a new {@link PrometheusCollector} to be registered with a {@link
+   * io.opentelemetry.sdk.metrics.SdkMeterProviderBuilder}.
+   */
+  public static PrometheusCollector create() {
+    return new PrometheusCollector();
+  }
```

Comment:
Could be worthwhile to accept a `CollectorRegistry` argument to this to allow registering to registries other than `CollectorRegistry.defaultRegistry`.

Reply:
sounds like a feature request!

Reply:
And not something I need myself so should wait